Storage/SQL/SQLiteHistory.swift

```diff
@@ -107,15 +107,15 @@ public class SQLiteHistory {
     let favicons: FaviconsTable<Favicon>
     let prefs: Prefs
 
-    required public init?(db: BrowserDB, prefs: Prefs) {
+    required public init(db: BrowserDB, prefs: Prefs) {
         self.db = db
         self.favicons = FaviconsTable<Favicon>()
         self.prefs = prefs
 
         // BrowserTable exists only to perform create/update etc. operations -- it's not
         // a queryable thing that needs to stick around.
         if !db.createOrUpdate(BrowserTable()) {
-            return nil
+            log.error("Unable to create or update browser table!")
```

Comment:
You could also have createOrUpdate return a tri-state value (success, failure, closed).